packages/deployer/subtasks/generate-router.js

```diff
@@ -18,13 +18,11 @@ subtask(
   'Reads deployed modules from the deployment data file and generates the source for a new router contract.'
 ).setAction(async (_, hre) => {
   const routerName = 'Router';
-  const routerPath = path.join(
-    relativePath(hre.config.paths.sources, hre.config.paths.root),
-    `${routerName}.sol`
-  );
+  const routerPath = path.join(hre.config.paths.sources, `${routerName}.sol`);
+  const relativeRouterPath = relativePath(routerPath, hre.config.paths.root);
```

Comment:
We were using the relative path to save the Router and if you used another hardhat.config.js file it didn't use its configured sources folder. Just the relative one from the running process.